src/main/java/org/kohsuke/github/GHIssueQueryBuilder.java

```diff
@@ -0,0 +1,163 @@
+package org.kohsuke.github;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GHIssueQueryBuilder extends GHQueryBuilder<GHIssue> {
+    private final GHRepository repo;
+    private List<String> labels = new ArrayList<>();
+
+    GHIssueQueryBuilder(GHRepository repo) {
+        super(repo.root());
+        this.repo = repo;
+    }
+
+    /**
+     * Milestone gh issue query builder.
+     *
+     * @param milestone
+     *            the milestone
+     * @return the gh issue request query builder
+     */
+    public GHIssueQueryBuilder milestone(String milestone) {
+        req.with("milestone", milestone);
+        return this;
+    }
+
+    /**
+     * State gh issue query builder.
+     *
+     * @param state
+     *            the state
+     * @return the gh issue query builder
+     */
+    public GHIssueQueryBuilder state(GHIssueState state) {
+        req.with("state", state);
+        return this;
+    }
+
+    /**
+     * Assignee gh issue query builder.
+     *
+     * @param assignee
+     *            the assignee
+     * @return the gh issue query builder
+     */
+    public GHIssueQueryBuilder assignee(String assignee) {
+        req.with("assignee", assignee);
+        return this;
+    }
+
+    /**
+     * Creator gh issue query builder.
+     *
+     * @param creator
+     *            the creator
+     * @return the gh issue query builder
+     */
+    public GHIssueQueryBuilder creator(String creator) {
+        req.with("creator", creator);
+        return this;
+    }
+
+    /**
+     * Mentioned gh issue query builder.
+     *
+     * @param mentioned
+     *            the mentioned
+     * @return the gh issue query builder
+     */
+    public GHIssueQueryBuilder mentioned(String mentioned) {
+        req.with("mentioned", mentioned);
+        return this;
+    }
+
+    /**
+     * Labels gh issue query builder.
+     *
+     * @param label
+     *            the labels
+     * @return the gh issue query builder
+     */
+    public GHIssueQueryBuilder label(String label) {
+        if (label != null) {
+            labels.add(label);
+        }
```

Comment:
No super efficient, but simplifies the `list()` method. 
```suggestion
        if (label != null && label.trim() != "") {
            labels.add(label);
            req.put("labels", labels.stream().collect(Collectors.joining(",")));
        }
```